src/runtime/driver/mod.rs

```diff
@@ -122,6 +122,21 @@ impl Driver {
         ))
     }
 
+    pub(crate) fn submit_op_2(&mut self, sqe: squeue::Entry) -> usize {
```

Comment:
I think this function should be labelled `unsafe` and each ancestor becomes `unsafe` too until reaching `submit()` and there the rationale given for being safe is the unsafe contract is maintained by building the  squeue Entry per lifetime requirements. So then the unsafe chain can be seen going all the way back into each op. 

I see so many bugs and useless questions raised because people just cut-n-paste unsafe calls without ever reading into them and then writing their own responses at the call site.

Reply:
I agree